test/unit/helpers/setup.js

```diff
@@ -23,9 +23,23 @@
 
 import td from 'testdouble';
 
-// Returns a foundation configured to use a mock object with the same api as a default adapter,
-// as well as that adapter itself.
+/**
+ * Returns a foundation configured to use a mock object with the same API as a default adapter,
+ * as well as that adapter itself.
+ * The trailing `.` in the `@param` type below is intentional: It indicates a reference to the class itself instead of
+ * an instance of the class.
+ * See https://youtrack.jetbrains.com/issue/WEB-10214#focus=streamItem-27-1305930-0-0
+ * @param {!MDCFoundation.} FoundationClass
+ * @return {{mockAdapter: !Object, foundation: !MDCFoundation}}
+ */
 export function setupFoundationTest(FoundationClass) {
+  // Make code coverage happy by instantiating the class with no arguments.
```

Comment:
I'm confused... where were we missing coverage due to this? Judging by this comment it should currently be impossible for us to have 100% branch coverage on foundations, but that's clearly not the case...

Reply:
All I know is that Dialog code coverage was stuck at 19/20 until I added this line. 🤷‍♂️

Reply:
### Without this line:

![image](https://user-images.githubusercontent.com/409245/45107695-44a2ea80-b0ef-11e8-96c6-c708be8ebd02.png)

File |   | Statements |   | Branches |   | Functions |   | Lines |  
-- | -- | -- | -- | -- | -- | -- | -- | -- | --
constants.js |   | 100% | 3/3 | 100% | 0/0 | 100% | 0/0 | 100% | 3/3
foundation.js |   | 100% | 64/64 | 92.86% | 13/14 | 100% | 22/22 | 100% | 61/61
index.js |   | 100% | 46/46 | 100% | 6/6 | 100% | 31/31 | 100% | 45/45
util.js |   | 100% | 6/6 | 100% | 2/2 | 100% | 4/4 | 100% | 5/5

---

### With this line:

![image](https://user-images.githubusercontent.com/409245/45107765-7ae06a00-b0ef-11e8-844e-a7ce3457720b.png)

File |   | Statements |   | Branches |   | Functions |   | Lines |  
-- | -- | -- | -- | -- | -- | -- | -- | -- | --
constants.js |   | 100% | 3/3 | 100% | 0/0 | 100% | 0/0 | 100% | 3/3
foundation.js |   | 100% | 64/64 | 100% | 14/14 | 100% | 22/22 | 100% | 61/61
index.js |   | 100% | 46/46 | 100% | 6/6 | 100% | 31/31 | 100% | 45/45
util.js |   | 100% | 6/6 | 100% | 2/2 | 100% | 4/4 | 100% | 5/5

Reply:
I just ran coverage with/without this line and it only affects branch coverage in dialog, line-ripple, and list. I'll see if I can figure out why, but I am inclined to leave it out of this PR.

Reply:
I figured out why 3 specific components have diffs with this line. It's because for some reason those components are using a default parameter in their foundation constructors, whereas all other foundations use an `Object.assign` within the constructor instead. I'm not sure what prompted this deviation, as the `Object.assign` pattern existed much earlier.

Reply:
#3522 removes the default parameters from `line-ripple` and `list`.